kong/db/strategies/cassandra/connector.lua

```diff
@@ -143,6 +143,7 @@ function CassandraConnector.new(kong_config)
     timeout_read              = kong_config.cassandra_timeout,
     max_schema_consensus_wait = kong_config.cassandra_schema_consensus_timeout,
     ssl                       = kong_config.cassandra_ssl,
+    encryption_protocol       = kong_config.cassandra_ssl_encryption_protocol,
```

Comment:
Specifying a protocol here will indeed make lua-cassandra's LuaSec fallback use the specified TLS version. *However*, the LuaSec fallback is only used for OpenResty phases that do not support cosockets (https://github.com/openresty/lua-nginx-module#cosockets-not-available-everywhere). The TLS version of cosocket connections is determined by the [lua_ssl_protocols](https://github.com/openresty/lua-nginx-module#lua_ssl_protocols) directive instead, so this will only work for the init phases and few others were the LuaSec fallback is used, and not in the proxy path.

Reply:
@thibaultcha thanks for pointing out the phases where cosockets are not available. I was not aware of this. Follow up question for you. If LuaSec covers the phases where cosockets are not implemented, and where cosockets are available, the TLS version defaults to a list `lua_ssl_protocols SSLv3 TLSv1 TLSv1.1 TLSv1.2`, doesn't that then mean that we effectively have the capability to utilize (by configuration in this PR) any supported TLS version across 100% of the phases? If that is true, then it addresses the problem I'm trying to solve, where for instance in the init phase, we cannot control the TLS version being used by LuaSec so connections fail when TLSv1.2 is required.

Reply:
It seems to me as if this new property is trying to solve two distinct use-cases at once:
A. Supporting TLS 1.2 in non-cosocket phases (FTI-2249)
B. Introducing a way to configure the TLS version to use between Kong & Cassandra

While this new property does solve A. (upon user intervention to update its default), it does not guarantee B. across *all* connections opened with Cassandra. As of today (with and without this patch), `cassandra_ssl_encryption_protocol = tlsv1` and a different `lua_ssl_protocols` directive could already cause disparities between cosocket and LuaSec connections.
Besides, even if we were to try and specify `lua_ssl_protocols`, we would be affecting all workers' cosockets, which could break other connections in the gateway or established by plugins to other services. In order to fine-grain Kong <-> C* TLS connections specifically, I think we'd want to rely on [tcpsock:tlshandshake()](https://github.com/openresty/lua-nginx-module/pull/1602) to add a new option (this OpenResty patch is bundled with Kong).

Therefore in light of the above, it does not seem like a good time to address B. at the moment. 

However, maybe we could still fix A. by:
1. Making TLS 1.2 the new hard-coded default in lua-cassandra
2. Having lua-cassandra fallback to TLS 1.1 for backwards-compatibility (maybe by way of catching an invalid protocol error, or maybe by looking at newer LuaSec APIs?)

This way we'd still allow non-cosocket phases to default to TLS 1.2 much like cosockets. Later, we may revisit use-case B., which is about _specifically configuring Kong to Cassandra connections_; since I don't believe it to be a popular ask (when we distinguish it from A. that is).